pkg/chartverifier/checks/pyxis.go

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2021 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package checks
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+var pyxisBaseUrl = "https://catalog.redhat.com/api/containers/v1/repositories"
+
+type RepositoriesBody struct {
+	PyxisRepositories []PyxisRepository `json:"data"`
+}
+
+type PyxisRepository struct {
+	Id          string `json:"_id"`
+	Repository  string `json:"repository"`
+	VendorLabel string `json:"vendor_label"`
+	Registry    string `json:"registry"`
+}
+
+type RegistriesBody struct {
+	PyxisRegistries []PyxisRegistry `json:"data"`
+}
+
+type PyxisRegistry struct {
+	Id           string               `json:"_id"`
+	Repositories []RegistryRepository `json:"repositories"`
+}
+
+type RegistryRepository struct {
+	Registry   string          `json:"registry"`
+	Repository string          `json:"repository"`
+	Tags       []RepositoryTag `json:"tags"`
+}
+
+type RepositoryTag struct {
+	Digest string `json:"manifest_schema1_digest"`
+	Name   string `json:"name"`
+}
+
+func getImageRegistries(repository string) ([]string, error) {
+	var err error
+	var registries []string
+
+	requestFilter := fmt.Sprintf("filter=repository==%s", repository)
+	requestUrl := fmt.Sprintf("%s?%s", pyxisBaseUrl, requestFilter)
```

Comment:
let's use `req.URL.Query()` method instead of manually crafting proper GET url.

Reply:
Done, not sure it makes the code better but then one of my previous jobs was implementing the servlet spec so very familiar with request format. Changed as requested.